lib/ruby_saml/idp_metadata_parser.rb

```diff
@@ -392,15 +393,13 @@ def certificates
 
       # @return [String|nil] the fingerpint of the X509Certificate if it exists
       #
-      def fingerprint(certificate, fingerprint_algorithm = RubySaml::XML::Document::SHA256)
-        @fingerprint ||= begin
-          return unless certificate
+      def fingerprint(certificate, fingerprint_algorithm = RubySaml::XML::Crypto::SHA256)
```

Comment:
This method takes arguments so it can't be memoized (can't do `@fingerprint ||= ...` here)